src/main/scala/collection/immutable/LazyListExtensions.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection.immutable
+
+object LazyListExtensions {
+
+  implicit class LazyListCycle[T](ll: LazyList[T]) {
```

Comment:
With that name, we're going to need a separate `implicit class` for every extension that we add to `LazyList`. I would name this `LazyListOps` or `LazyListExtensionsOps`.

Reply:
we also have to be mindful of the fact that in Scala 2, the name of that implicit class will shadow any other implicit class with that name, from any library

Reply:
I went with `NextLazyListExtensions`, but I can change it again based on however the discussion at #24 turns out